packages/module/src/type-utils.ts

```diff
@@ -10,3 +12,12 @@ export type DisallowCharacters<
 export type DiscriminatedArgs<TBoolDiscriminatorKey extends string, TArgs> =
   | ({ [key in TBoolDiscriminatorKey]: true } & TArgs)
   | { [key in TBoolDiscriminatorKey]?: false };
+
+export type MergedArgs<
+  A extends Partial<Record<TableFeature, object>>,
+  B extends Partial<Record<TableFeature, object>>,
+  TIncludedFeatures extends TableFeature = TableFeature
+> = Omit<A, TableFeature> &
+  Omit<B, TableFeature> & {
+    [key in TIncludedFeatures]: [A, B] extends [undefined, undefined] ? undefined : A[key] & B[key];
```

Comment:
This type is tough to read, and ideally undefined is not something extended. Maybe a comment describing what is going on here would help at least.

Reply:
Yeah, this could use another look. The thing I was trying to express here is that if A or B here both don't have an object for a feature (the user omitted it to disable that feature) we shouldn't try to index A[key] or B[key] there and instead the property for that feature should be undefined. It may be that just `A[key] & B[key]` would work here, I'll try that. I think I added this weird conditional type in debugging.

Reply:
Even without the conditional that might have been added in debugging, I think a comment to this effect would be helpful for future reference to the type.

Reply:
Addressed